src/tasks.js

```diff
@@ -1,19 +1,167 @@
+            // delete selected item from dom
+            li.remove();
+            Tasks.deleteTask(task.index);
+            Tasks.setLocal();
+          });
+        });
+        ul.appendChild(li);
+      });
+      const li = document.createElement('li');
+      const p = document.createElement('p');
+      p.innerText = 'Clear all completed';
+      li.appendChild(p);
+      li.classList.add('d-flex', 'justify-content-around', 'bg', 'cursor');
+      li.addEventListener('click', () => {
+        Tasks.deleteCompleted();
+      });
+      ul.appendChild(li);
+    }
+
+    static getLocal() {
+      // get local storage
+      const tasks = JSON.parse(localStorage.getItem('tasks'));
+      if (tasks) {
+        Tasks.tasks = tasks;
+      }
+      Tasks.showTasks();
+    }
+
+    addTasks() {
+      // eslint-disable-next-line no-return-assign
+      Tasks.tasks.forEach((e, i) => e.index = i + 1);
+      // push from the last index deleted
+      Tasks.tasks.push(this);
+      // remove repeated tasks
+      Tasks.tasks = Tasks.tasks.filter((task, index, self) => index === self.findIndex((t) => (
+        t.description === task.description
+      )));
+      Tasks.setLocal();
+      // add Tasks to dom
+      Tasks.showTasks();
+    }
+
+    // delete task
+    static deleteTask(index) {
+      // delete from an array in order
+      if (index === 1 || this.tasks.length === 1) {
+        Tasks.tasks.splice(0, 1);
+      } else if (index === this.tasks.length) {
+        Tasks.tasks.splice(this.tasks.length - 1, 1);
+      } else {
+        Tasks.tasks.splice(index, 1);
+      }
+      // set local
+      Tasks.setLocal();
+    }
+
+    static editTask(index) {
+      // const li = document.querySelector(`li:nth-child(${index})`);
+      const description = document.querySelectorAll('p');
+      description.forEach((desc) => {
+        desc.setAttribute('contenteditable', 'true');
+        desc.addEventListener('blur', () => {
+          desc.setAttribute('contenteditable', 'false');
+          Tasks.tasks[index - 1].description = desc.innerText;
+          Tasks.setLocal();
+          Tasks.showTasks();
+        });
+        desc.addEventListener('keypress', (e) => {
+          if (e.key === 'Enter') {
+            desc.setAttribute('contenteditable', 'false');
+            Tasks.tasks[index - 1].description = desc.innerText;
+            Tasks.setLocal();
+            Tasks.showTasks();
+          }
+        });
+      });
+    }
+
+    // delete all tasks
+    static deleteAllTasks() {
+      Tasks.tasks = [];
+      Tasks.setLocal();
+    }
+
+    static deleteCompleted() {
+      const domStay = document.querySelectorAll('.completed');
+      const stay = Tasks.tasks.filter((task) => task.completed === false);
+      Tasks.tasks = stay;
+      domStay.forEach((domEl) => {
+        domEl.remove();
+      });
+      this.setLocal();
+    }
+}
+
+refresh.addEventListener('click', () => {
+  Tasks.deleteAllTasks();
+  ul.remove();
+});
```

Comment:
- [ ] **OPTIONAL**
Extra credit for implementing a use for the refresh button. However, it seems that you're removing the whole list element from the app when this event is fired, and this is preventing users from adding tasks after! It'd be great if you worked on that either 😄 😉

Reply:
Hello I already resolved all the required changes you highlighted, I only left a few of the optional to do at my free time, you can double check